### PR TITLE
Fix issue 17217 - std.net.isemail.isEmail doesn't work with non char …

### PR DESCRIPTION
**Pull request by JackStouffer**
…arrays

A simple fix with a large diff.

### CODE REVIEW

File: std/net/isemail.d [hunk 1]
```diff
@@ -75,6 +71,11 @@ if (isSomeChar!(Char))
     import std.uni : isNumber;
 
     alias tstring = const(Char)[];
+    alias TstringToken = Token!(Char);
```
Comment (DmitryOlshansky) [inline std/net/isemail.d]: A simpler change is to rename current Token --> TokenImpl, and then add `alias Token = TokenImpl!Char`

**Comment by JackStouffer** [comment]
@DmitryOlshansky Fixed

**Comment by JackStouffer** [comment]
Auto-merge toggled on

**Comment by JackStouffer** [comment]
I removed the label because the auto-merge label does not prioritize the PR in the auto-tester and so it was taking too long.

**Comment by wilzbach** [comment]
FYI
- no need to remove the label (closed PRs are simply not considered) 
Known issue (https://github.com/dlang-bots/dlang-bot/issues/49) - I even wrote a scraper for the auto-tester, but I still have hopes that we will see an auto-tester update ;-)

**Pull request merged**